feat(arithmetic): Adds equations as possible columns to events v2

wmak opened the pull request:
- This adds the ability to use equations as columns in eventsv2
- This doesn't add the ability to query for equations yet or to use them
  on the stats endpoint yet

wmak committed:
feat(arithmetic): Adds equations as possible columns to events v2

- This adds the ability to use equations as columns in eventsv2
- This doesn't add the ability to query for equations yet or to use them
  on the stats endpoint yet
-

Zylphrex commented on src/sentry/discover/arithmetic.py:
```diff
@@ -53,6 +57,15 @@ def validate(self) -> None:
+        """ Convert this tree of Operations to the equivalent snuba json """
+        lhs = self.lhs.to_snuba_json() if isinstance(self.lhs, Operation) else self.lhs
+        rhs = self.rhs.to_snuba_json() if isinstance(self.rhs, Operation) else self.rhs
+        result = [self.operator, [lhs, rhs]]
```
Since it's only allowing for 1 operation right now this should be okay but something to be wary of when allowing for more deeply nested expressions is the following

```
["plus", ["transaction.duration", ["plus", ["spans.http", "measurements.lcp"]]]]
```

Because snuba _may_ interpret `transaction.duration` as a function call.

wmak replied:
:+1: In testing I think this worked with fields but good reminder if I see this fail 🙃

Zylphrex reviewed (approved):
Just one thing to be wary of regarding the JSON syntax but otherwise LGTM

wmak committed:
fix: Handle equations being empty

wmak committed:
fix: Fixing tests

- bad timestamp for the event_data
- forgot to reduce max_operators back to 1 after testing

wmak committed:
Merge branch 'master' into wmak/feat/equations-column
